pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -567,6 +569,14 @@ func generateVirtualHostDomains(service *model.Service, listenerPort int, port i
 	if len(svcAddr) > 0 && svcAddr != constants.UnspecifiedIP {
 		domains = appendDomainPort(domains, svcAddr, port)
 	}
+
+	// handle dual stack's extra address when generating the virtualHost domains
+	// assumes that conversion is stripping out the DefaultAddress from ClusterVIPs
+	extraAddr := service.GetExtraAddressesForProxy(node)
```

Comment:
I recommended initially, and continue to believe, the concept of "Extra" addresses is flawed. It just gives the codebase a super easy way to forget to handle the fact we can have multiple addresses.

Reply:
Also it seems like Get**Extra**AddressesForProxy is not really "Extra" addresses, but rather **all** addresses. So not sure how  are not getting duplicate addresses here? Maybe we are

Reply:
service conversion from a k8s svc to model strips it out right now. Otherwise it would duplicate, you are correct. Not sure long term that is actually the way to go and maybe its just better to align with K8s (and potentially remove the "extra" nomenclature while we're at it).

Reply:
> Also it seems like Get**Extra**AddressesForProxy is not really "Extra" addresses, but rather **all** addresses. So not sure how are not getting duplicate addresses here? Maybe we are

I was (partially) mistaken in my initial response to this. There are other code paths that create services which this method needs to be able to handle and those do include duplicate IPs so the content of the field isn't consistent across the project. Sometimes we do get extra IPs and sometimes we don't depending on how the service was created.